types/src/validator_set.rs

```diff
@@ -78,13 +92,71 @@ impl ValidatorSetExt for Set {
             }
         }
 
+        Err(VerificationError::NotEnoughVotingPower(
+            tallied_voting_power,
+            voting_power_needed,
+        ))?
+    }
+
+    fn verify_commit_light_trusting(
+        &self,
+        chain_id: &chain::Id,
+        commit: &block::Commit,
+        trust_level: TrustLevelRatio,
+    ) -> Result<()> {
+        let mut seen_vals = HashMap::<usize, usize>::new();
+        let mut tallied_voting_power = 0;
+
+        let voting_power_needed = trust_level.total_power_needed(self.total_voting_power())?;
+
+        for (idx, commit_sig) in commit.signatures.iter().enumerate() {
+            let (val_id, signature) = match commit_sig {
+                CommitSig::BlockIdFlagCommit {
+                    validator_address,
+                    signature: Some(ref sig),
+                    ..
+                } => (validator_address, sig),
+                CommitSig::BlockIdFlagCommit { .. } => {
+                    bail_verification!("No signature in CommitSig");
+                }
+                // not commiting for the block
+                _ => continue,
+            };
+
+            let Some((val_idx, validator)) = find_validator(self, val_id) else {
+                continue;
+            };
+
+            if let Some(prev_idx) = seen_vals.get(&val_idx) {
+                bail_verification!("Double vote from {val_id} ({prev_idx} and {idx}");
+            }
+
+            seen_vals.insert(val_idx, idx);
+
+            let vote_sign = commit.vote_sign_bytes(chain_id, idx)?;
+            validator.verify_signature::<Verifier>(&vote_sign, signature)?;
+
+            tallied_voting_power += validator.power();
+
+            if tallied_voting_power > voting_power_needed {
+                return Ok(());
+            }
+        }
+
         Err(Error::NotEnoughVotingPower(
             tallied_voting_power,
             voting_power_needed,
-        ))
+        ))?
```

Comment:
this `?` changes the type of this `Err` to `Err::<Result<()>, Error>`. I've been wrapping my head how does it work for a good while. If you don't mind, I'd prefer this (and one from previous `verify_commit...`) without `?` so that this will be of type `Err::<(), Error>`

Reply:
Sorry it was a leftover when I was experimenting with errors. I will remove it.